runway/templates/k8s-flux-repo/flux.tf/main.tf

```diff
@@ -0,0 +1,468 @@
+        }
+
+        volume {
+          name = "git-config"
+
+          config_map {
+            name         = "flux-git-config"
+            default_mode = "0644"
+          }
+        }
+
+        automount_service_account_token = true
+
+        container {
+          name  = "flux"
+          image = "${aws_ecr_repository.flux_with_awscli.repository_url}:${local.flux_version}"
+          args  = [
+            "--memcached-service=", 
+            "--ssh-keygen-dir=/var/fluxd/keygen", 
+            "--git-url=$(GIT_URL)", 
+            "--git-branch=master", 
+            "--git-path=namespaces,workloads", 
+            "--git-label=flux", 
+            "--git-user=Flux", 
+            "--git-email=Flux", 
+            "--listen-metrics=:3031",
+            "--sync-garbage-collection"]
+
+          port {
+            container_port = 3030
+            protocol       = "TCP"
+          }
+
+          env {
+            name = "GIT_URL"
+
+            value_from {
+              config_map_key_ref {
+                name = "flux-git-config"
+                key  = "giturl"
+              }
+            }
+          }
+
+          resources {
+            requests {
+              cpu    = "50m"
+              memory = "64Mi"
+            }
+          }
+
+          volume_mount {
+            name       = "git-key"
+            read_only  = true
+            mount_path = "/etc/fluxd/ssh"
+          }
+
+          volume_mount {
+            name       = "git-keygen"
+            mount_path = "/var/fluxd/keygen"
+          }
+
+          volume_mount {
+            name       = "ssh-config"
+            mount_path = "/root/.ssh"
+          }
+
+          volume_mount {
+            name       = "git-config"
+            mount_path = "/root/.gitconfig"
+            sub_path   = "gitconfig"
+          }
+
+          liveness_probe {
+            http_get {
+              path   = "/api/flux/v6/identity.pub"
+              port   = "3030"
+              scheme = "HTTP"
+            }
+
+            initial_delay_seconds = 5
+            timeout_seconds       = 5
+            period_seconds        = 10
+            success_threshold     = 1
+            failure_threshold     = 3
+          }
+
+          readiness_probe {
+            http_get {
+              path   = "/api/flux/v6/identity.pub"
+              port   = "3030"
+              scheme = "HTTP"
+            }
+
+            initial_delay_seconds = 5
+            timeout_seconds       = 5
+            period_seconds        = 10
+            success_threshold     = 1
+            failure_threshold     = 3
+          }
+
+          termination_message_path = "/dev/termination-log"
+          image_pull_policy        = "IfNotPresent"
+        }
+
+        restart_policy                   = "Always"
+        termination_grace_period_seconds = 30
+        dns_policy                       = "ClusterFirst"
+        service_account_name             = "${local.flux_service_account}"
```

Comment:
I think the service account can be dynamically referenced here, allowing the manual dependency on the service account resource to be removed (e.g. `"${kubernetes_service_account.flux_service_account.metadata.0.name}"`)

(I'll admit I'm not sure this is a worthwhile optimization but I have a preference towards minimizing the manual dependency listings)